src/gr11review/part2/Utility.java

```diff
@@ -1,5 +1,405 @@
+          intDiagonal[intCount][intCount1] = 2;
+        }
+      }
+    }
+    
+    //print out the array to diaonalOut text file
+    for(intCount = 0; intCount < n; intCount++){
+      for(intCount1 = 0; intCount1 <n; intCount1++){
+        Writer.print(intDiagonal[intCount][intCount1]);
+        if(intCount1 != n - 1){
+          Writer.print(", ");
+        }
+      }
+      Writer.println();
+    }
+    Writer.close();
+  } 
+
+
+    /**
+    *Create a method that for patterns like "zip" and "zap" in the string -- length-3, starting with 'z' and *ending with 'p'. Return a string where for all such words, the middle letter is gone, so "zipXzap"  *yields *"zpXzp".
+    *
+    * @param strZipZap - the string that contains the patterns with the z's and p's
+    * @return strZP - the string without the letter between z and p
+    * @author: E. Aroozoo
+    */
+
+    public static String zipZap(String strZipZap) {
+
+      //Variables
+      String strZP;
+      int intCounter;
+      int intSentenceLength;
+
+      //Setting Variables
+      strZP = "";
+      intSentenceLength = strZipZap.length();
+
+      //If Sentence less than 2 letters, send back sentence (does not contain z*p pattern)
+      if(intSentenceLength >= 3) {
+         //Counter for sentence, checks if there is z and p
+        for(intCounter = 0;intCounter < intSentenceLength; intCounter++) {
+          strZP += strZipZap.substring(intCounter, intCounter + 1);
+            if(intCounter > 0 && intCounter < intSentenceLength - 1) {
+
+              //Finds out if there is a z and p pattern and collects the string
+              if(strZipZap.charAt(intCounter - 1) == 'z' && strZipZap.charAt(intCounter+1) == 'p')
+              strZP = strZP.substring(0,strZP.length() - 1);
+            }
+        }
+
+        //Return string removing the letter between "z" and "p"
+        return strZP;
+      } else {
+        return strZP;
+      }
+    }
+    /**
+    *Write a method longestWord(String filenametxt) such that given the name of a file filenametxt that *contains a single word on each line, returns the longest word in the file.
+    *
+    * @param filename - the file that contains all the words
+    * @return strWord - the string that is the longest in the file
+    * @author: E. Aroozoo
+    */
+
+    public static String longestWord(String filenametxt) throws IOException {
+      BufferedReader filename = new BufferedReader(new FileReader(filenametxt));
+
+      //Variables
+      String strWord;
+      String strLongestWord;
+      int intWordLength;
+      int intLongestWordLength;
+
+      //Setting Variables (Putting the data from the file into the variable)
+      strWord = filename.readLine();
+      strLongestWord = filename.readLine();
+
+
+      //Comparing the next word to see if it is longer than the current word
+      while(strWord != null) {
+        intWordLength = strWord.length();
+        intLongestWordLength = strLongestWord.length();
```

Comment:
@EngelbertEA In the case where there is only one word in the text file, the method fails here.

`java.lang.NullPointerException: Cannot invoke "String.length()" because "strLongestWord" is null`